### PR TITLE
Move repos and images into architecture specific subdirs

### PR DESCRIPTION
**Pull request by dirkmueller**
This makes it much easier to fix the mkcloud/qa_crowbarsetup.sh
scripts. compatibility symlinks have been added manually on the server
until the other changes have been merged

### CODE REVIEW

**Comment by bmwiedemann** [comment]
+1 makes the dir structure more homogenous

**Comment by vuntz** [comment]
Hrm, this seems to be the old sync script... Would it be possible to use the updated version? :-) (it's used already on clouddata)

**Comment by dirkmueller** [comment]
@vuntz this is the new rewritten script (I had already fixed that in the old script, this is the forward port to the rewritten one).

File: scripts/clouddata/syncSLErepos [hunk 1]
```diff
@@ -152,9 +153,9 @@ for version in 4 5; do
     mount_and_rsync "/mnt/dist/ibs/Devel\:/Cloud\:/$version\:/Staging/images/iso/SUSE-CLOUD-$version-x86_64-Build[0-9][0-9][0-9][0-9]-Media1.iso" SUSE-Cloud-$version-devel-staging
 
     $rsync /mnt/euklid/mirror/SuSE/zypp-patches.suse.de/x86_64/update/SUSE-CLOUD/$version-POOL/ /srv/nfs/repos/SUSE-Cloud-$version-Pool/
-    $rsync /mnt/euklid/mirror/SuSE/build-ncc.suse.de/SUSE/Updates/SUSE-CLOUD/$version/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-$version-Updates/
-    if test -d /mnt/dist/ibs/SUSE\:/Maintenance\:/Test\:/Cloud\:/$version\:/$arch/update/; then
-        $rsync /mnt/dist/ibs/SUSE\:/Maintenance\:/Test\:/Cloud\:/$version\:/$arch/update/ /srv/nfs/repos$repo_subdir/SUSE-Cloud-$version-Updates-test/
+    $rsync $buildncc/SUSE/Updates/SUSE-CLOUD/$version/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-$version-Updates/
+    if test -d $ibsmaint/Cloud\:/$version\:/$arch/update/; then
+        $rsync $ibsmaint/Cloud\:/$version\:/$arch/update/ /srv/nfs/repos/$arch/SUSE-Cloud-$version-Updates-test/
```
Comment (vuntz) [inline scripts/clouddata/syncSLErepos]: Looks like a bug in the old script, but there's no $arch here in this loop.

Which makes me wonder: do you want to also move the repos to the arch-specific subdirs for 4 and 5? (If yes, it's missing here)

File: scripts/clouddata/syncSLErepos [hunk 2]
```diff
@@ -165,9 +166,12 @@ echo ================== Updating Cloud 5 for SLE12
...
 mount_and_rsync "/mnt/dist/ibs/Devel:/Cloud:/5:/Staging/images/iso/SUSE-SLE12-CLOUD-5-COMPUTE-x86_64-Build[0-9][0-9][0-9][0-9]-Media1.iso" SUSE-Cloud-5-SLE-12-devel-staging
-$rsync /mnt/euklid/mirror/SuSE/build.suse.de/SUSE/Products/12-Cloud-Compute/5/x86_64/product/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Pool/
-$rsync /mnt/euklid/mirror/SuSE/build.suse.de/SUSE/Updates/12-Cloud-Compute/5/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Updates/
-$rsync /mnt/dist/ibs/SUSE\:/Maintenance\:/Test\:/12-Cloud-Compute\:/5\:/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Updates-test/
+$rsync $buildsuse/SUSE/Products/12-Cloud-Compute/5/x86_64/product/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Pool/
+$rsync $buildsuse/SUSE/Updates/12-Cloud-Compute/5/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Updates/
+$rsync $ibsmaint/12-Cloud-Compute\:/5\:/x86_64/update/ /srv/nfs/repos/SUSE-Cloud-5-SLE-12-Updates-test/
+
+# mirror aarch64 SP1 builds
+$rsync /mnt/dist/ibs/Devel:/ARM:/SLE-12-SP1:/Update/images/repo/SLE-12-SP1-Server-POOL-aarch64-*-Media1/ /srv/nfs/repos/aarch64/SLES12-SP1-Pool
```
Comment (vuntz) [inline scripts/clouddata/syncSLErepos]: Can this be moved up to be with the code about SLES? I guess we should change the code around line 73 to not do a rsync first from the SUSE namespace, and then a second one from Devel:ARM.

File: scripts/clouddata/syncSLErepos (hunk 1 above)
Comment (dirkmueller) [inline scripts/clouddata/syncSLErepos]: no, I wanted to do it only for cloud 6 onwards.

File: scripts/clouddata/syncSLErepos (hunk 2 above)
Comment (dirkmueller) [inline scripts/clouddata/syncSLErepos]: @vuntz there is no sync from the suse namespace (this is the code about the repo (pool) sync, not the image sync).

Comment (vuntz) [inline scripts/clouddata/syncSLErepos]: There is, look at line 73 :-)

Comment (dirkmueller) [inline scripts/clouddata/syncSLErepos]: Its not executed, see line 56:

[ "$arch" = "aarch64" -a $servicepack -eq 1 ] && continue

**Comment by bmwiedemann** [comment]
still +1 if it works.

**Pull request merged**